### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by lodenos**
Merge ...

### CODE REVIEW

**Comment by tomt1664** [comment]
I'm not quite sure the reason that the `policytransactions.py` test is failing. The test fails at line 128, which is where the policy transaction is sent, which gets blocked with a `JSONRPC error: 64: blocked-asset-issuance-txn` error. 

This transaction has only outputs of TX_MULTISIG, so should not be blocked by the IsRedemption or IsValidBurn checks. But in any case, these checks only need to be done if `!IsPolicy` is true.

File: src/validation.cpp [hunk 1]
```diff
@@ -1193,32 +1183,32 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
...
+        if (fRequireFreezelistCheck) {
+          if (IsFreezelisted(tx, view)) {
+            if (fEnableBurnlistCheck) {
+              if (!IsBurnlisted(tx, view))
+                return state.DoS(0, false, REJECT_NONSTANDARD, "freezelist-no-burnlist-address");
+            } else
+              return state.DoS(0, false, REJECT_NONSTANDARD, "freezelist-address");
+          }
+          if (!IsRedemption(tx))
+            return state.DoS(0, false, REJECT_NONSTANDARD, "blocked-asset-issuance-txn");
```
Comment (nkostoulas) [inline src/validation.cpp]: need to rename this error

File: src/validation.cpp [hunk 2]
```diff
@@ -1193,32 +1183,32 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
...
+        }
+        if (fEnableBurnlistCheck && !IsValidBurn(tx, view))
+          return state.DoS(0, false, REJECT_NONSTANDARD, "blocked-asset-issuance-txn");
```
Comment (nkostoulas) [inline src/validation.cpp]: And here as well. "blocked-asset-issuance-txn" is used for block issuance error below and thus not possible to distinguish in the failed unit-test

Comment (nkostoulas) [reply src/validation.cpp]: you still haven't changed this

File: qa/rpc-tests/validBurn.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  # Create Address
+  #=============================================================================
+  addr0 = "2dZRkPX3hrPtuBrmMkbGtxTxsuYYgAaFrXZ"
+  addr1 = node.getnewaddress()
+  addr2 = node.getnewaddress()
+  addr3 = node.getnewaddress()
+  #=============================================================================
+  # Add address to FreezeList
+  #=============================================================================
+  node.addtofreezelist(addr1)
```
Comment (nkostoulas) [inline qa/rpc-tests/validBurn.py]: you are adding to freezelist not burnlist

File: qa/rpc-tests/validBurn.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  fee = Decimal('0.0001')
+  # Make Inputs
+  inputs = [{
+    "txid": unspent[0]["txid"],
+    "vout": unspent[0]["vout"],
+    "nValue": unspent[0]["amount"]
+  }]
+  # Make Outputs
+  outputs = {
+    addr3 : unspent[0]["amount"] - 3 - fee,
```
Comment (nkostoulas) [inline qa/rpc-tests/validBurn.py]: there's no null address on the outputs and that is why the test passes

File: src/policy/policy.cpp [hunk 5]
```diff
@@ -186,28 +186,56 @@ bool IsRedemption(CTransaction const &tx) {
   return true;
 }
 
+#define P cout << __func__ << " : " << __LINE__ << endl
```
Comment (nkostoulas) [inline src/policy/policy.cpp]: spam

File: qa/rpc-tests/validBurn.py (hunk 4 above)
Comment (tomt1664) [reply qa/rpc-tests/validBurn.py]: Yes - the null address is a real address, not an op_return. For regtest use: 2dZRkPX3hrPtuBrmMkbGtxTxsuYYgAaFrXZ for null address.

**Comment by tomt1664** [comment]
The single Travis failure may be a problem with my test (policytransactions.py) - I'll take a look.

**Pull request closed without merging**